stdlib/jsonutils/src/test/java/org/ballerinalang/stdlib/jsonutils/JsonUtilsTest.java

```diff
@@ -64,30 +68,161 @@ public void testFromTableFunction() {
+                        "\"@discount\":\"22%\"}, \"Count\":{\"Count\":\"7\", \"@xmlns\":\"example.com\"}}, " +
+                        "\"@xmlns\":\"example.com\"}, \"\\n        \"], \"@xmlns\":\"example.com\"}, " +
+                        "\"\\n        \", " +
+                        "{\"Address\":[\"\\n            \", " +
+                        "{\"StreetAddress\":\"20, Palm grove, Colombo 3\"}, \"\\n            \", " +
+                        "{\"City\":\"Colombo\"}, \"\\n            \", {\"Zip\":\"00300\"}, \"\\n            \", " +
+                        "{\"Country\":\"LK\"}, \"\\n        \"], \"@xmlns\":\"\"}, \"\\n    \"], " +
+                        "\"@xmlns\":\"example.com\", \"@attr\":\"attr-val\", \"@ns:attr\":\"ns-attr-val\", " +
+                        "\"@xmlns:ns\":\"ns.com\"}}");
     }
 
     @Test
-    public void testFromTableFunction2() {
-        BValue[] returns = BRunUtil.invoke(result, "testFromTable2");
+    public void testUsingConvertedJsonValue() {
+        BValue[] res = BRunUtil.invoke(result, "testUsingConvertedJsonValue");
+        Assert.assertEquals(res[0].stringValue(), "BCD:ZZZ");
+    }
+
+    @Test
+    public void testXmlToJsonToPInfo() {
+        BValue[] res = BRunUtil.invoke(result, "testXmlToJsonToPInfo");
+        Assert.assertEquals(res[0].stringValue(), "{name:\"Jane\", age:\"33\", gender:\"not-specified\"}");
+    }
+
+    @Test
+    public void testXMLWithEmptyChildren() {
+        BValue[] returns = BRunUtil.invoke(result, "testXMLWithEmptyChildren");
         Assert.assertNotNull(returns[0]);
-        Assert.assertEquals(returns[0].stringValue(),
-                "[{\"id\":1, \"name\":\"Mary\", \"salary\":300.5, \"permanent\":true, " +
-                        "\"dependents\":[\"Mike\", \"Rachel\"], " +
-                        "\"contact\":{" +
-                            "\"phone\":[445566, 778877], " +
-                            "\"address\":{\"number\":34, \"street\":\"Straford\"}, " +
-                        "\"emergency\":\"Stephen\"}}, " +
-                        "{\"id\":2, \"name\":\"John\", \"salary\":200.5, \"permanent\":false, " +
-                        "\"dependents\":[\"Kyle\"]," +
-                        " \"contact\":{" +
-                            "\"phone\":[6060606, 556644], \"address\":{\"number\":10, \"street\":\"Oxford\"}, " +
-                        "\"emergency\":\"Elizabeth\"}}, " +
-                        "{\"id\":3, \"name\":\"Jim\", \"salary\":330.5, \"permanent\":true, " +
-                        "\"dependents\":[], " +
-                        "\"contact\":{" +
-                            "\"phone\":[960960, 889889], " +
-                            "\"address\":{\"number\":46, \"street\":\"Queens\"}, " +
-                        "\"emergency\":\"Veronica\"}}]");
+        Assert.assertEquals(returns[0].stringValue(), "{\"foo\":{\"bar\":\"2\", \"car\":\"\"}}");
```

Comment:
Can we not assert these in Ballerina itself?

Reply:
And in Ballerina, shall we assert the value itself, without converting to string?

Reply:
You mean to create the complete json value and assert or assert specific bits?

Reply:
We can create the complete value and assert against it right?

Reply:
For simple json values like this one, we can do it, but I don't think it's a productive thing to do for bit complex cases like in L:77.

Reply:
Then maybe we can have a set of tests that cover the creation of new values, like conversion to arrays and maps, where we actually assert against the value and maybe a separate set to cover composite scenarios.

My concern with string representations is that we don't assert the actual type. For example, here this could be a string itself or the string representation of a JSON object, right? And this can go on for nested values too.